src/node_api.cc

```diff
@@ -655,6 +655,36 @@ v8::Local<v8::Object> CreateAccessorCallbackData(napi_env env,
   return cbdata;
 }
 
+// Pointer used to identify items wrapped by N-API. Used by FindWrapper and
+// napi_wrap().
+char napi_wrap_name[] = "N-API Wrapper";
+
+// Search the object's prototype chain for the wrapper object. Usually the
+// wrapper would be the first in the chain, but it is OK for other objects to
+// be inserted in the prototype chain.
+bool FindWrapper(v8::Local<v8::Object> obj,
```

Comment:
> So, if one module does napi_wrap(), then the other module will not recognize that this has happened, because it will have its own copy of a FunctionTemplate serving the same purpose.

It doesn't have to be.  FunctionTemplates can be per-isolate (don't need to be per-context) so if you manage to associate some state with the isolate, you should be good.

As to how to do that: you could use `v8::Isolate::SetData()` but that's a scarce resource (only 4 slots and one is already taken).

Perhaps a static `std::map<v8::Isolate*, State*>` guarded by a mutex?  You only need to look it up once per `napi_env` because you can cache it inside the env.  Lock contention shouldn't be an issue.